examples/BasicOperations/SearchForGoogleAdsFields.php

```diff
@@ -96,54 +96,80 @@ public static function main()
      * @param GoogleAdsClient $googleAdsClient the Google Ads API client
-     * @param string $artifactName the name of artifact to get its metadata
+     * @param string $namePrefix the name prefix to use in the query
      */
-    public static function runExample(GoogleAdsClient $googleAdsClient, string $artifactName)
+    public static function runExample(GoogleAdsClient $googleAdsClient, string $namePrefix)
     {
         $googleAdsFieldServiceClient = $googleAdsClient->getGoogleAdsFieldServiceClient();
-        // Searches for an artifact whose name is the same as the specified artifactName.
+        // Searches for all fields whose name begins with the specified namePrefix.
+        // A single "%" is the wildcard token in the Google Ads Query language.
         $query = "SELECT name, category, selectable, filterable, sortable, selectable_with, "
-            . "data_type, is_repeated WHERE name = '$artifactName'";
+            . "data_type, is_repeated WHERE name LIKE '%$namePrefix%'";
         $response = $googleAdsFieldServiceClient->searchGoogleAdsFields($query);
 
-        // Iterates over all rows and prints our the metadata of the returned artifacts
+        if (iterator_count($response->getIterator()) === 0) {
+            printf(
+                "No GoogleAdsFields found with a name that begins with %s.%s",
+                $namePrefix,
+                PHP_EOL
+            );
+            return;
+        }
+        // Iterates over all rows and prints our the metadata of each matching GoogleAdsField.
         foreach ($response->iterateAllElements() as $googleAdsField) {
             /** @var GoogleAdsField $googleAdsField */
+            printf("%s:%s", $googleAdsField->getName(), PHP_EOL);
             printf(
-                "An artifact named '%s' with category '%s' and data type '%s' %s selectable, %s "
-                . "filterable, %s sortable and %s repeated.%s",
-                $googleAdsField->getName(),
+                "  %-16s: %s%s",
+                "category:",
                 GoogleAdsFieldCategory::name($googleAdsField->getCategory()),
+                PHP_EOL
+            );
+            printf(
+                "  %-16s: %s%s",
+                "data type:",
                 GoogleAdsFieldDataType::name($googleAdsField->getDataType()),
-                self::getIsOrIsNot($googleAdsField->getSelectable()),
-                self::getIsOrIsNot($googleAdsField->getFilterable()),
-                self::getIsOrIsNot($googleAdsField->getSortable()),
-                self::getIsOrIsNot($googleAdsField->getIsRepeated()),
                 PHP_EOL
             );
+            printf(
+                "  %-16s: %s%s",
+                "selectable:",
+                $googleAdsField->getSelectable() ? 'true' : 'false',
+                PHP_EOL
+            );
+            printf(
+                "  %-16s: %s%s",
+                "filterable:",
+                $googleAdsField->getFilterable() ? 'true' : 'false',
```

Comment:
Is this necessary with booleans because otherwise %s will print them as a 1 or 0?

Reply:
Yes, PHP is weird with booleans. It'll just print 1 (for true) or nothing at all (for false) (not even 0).